src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -578,10 +584,12 @@ ResultWithMessage window_editor_object_selection_select_object(
     {
         for (const auto& sgEntry : item->SceneryGroupInfo.Entries)
         {
+            ObjectEntryDescriptor x;
             const auto selectionResult = window_editor_object_selection_select_object(++isMasterObject, flags, sgEntry);
             if (!selectionResult.Successful)
             {
                 _gSceneryGroupPartialSelectError = selectionResult.Message;
+                _gErrorSceneryGroupObjectEntryDescriptors.push_back(sgEntry);
```

Comment:
Why make this span two files to display the error? log messages can just be done at any point.

Reply:
The initial plan was to have the missing objects shown in the error popup when selecting a scenery group, but I was running into some trouble figuring out how to get that part done on the UI side. I had it in EditorObjectSelection.cpp just to verify I could get the object names from the object selection window. If it makes more sense to keep it all in EditorObjectSelectionSession.cpp, I can move it there instead.

Reply:
Just keep it to the one file for now producing the error immediately. We will eventually want to pass it to the ui but it really needs a better design to handle it nicely.

Reply:
This has been updated